src/modules/navigator/rtl.cpp

```diff
@@ -709,13 +714,12 @@ float RTL::calculate_return_alt_from_cone_half_angle(float cone_half_angle_deg)
 	return max(return_altitude_amsl, gpos.alt);
 }
 
-void RTL::calc_and_pub_rtl_time_estimate(const RTLState rtl_state)
+void RTL::calc_and_pub_rtl_time_estimate(const RTLState rtl_state, bool global_position_recently_updated)
```

Comment:
Having to pass a boolean to bypass most of the logic in a function is usually because of a design flaw. In that case, the problem is because the function does 2 main things: calculate *and* publish.

You could instead make a function that does only the calculation part if the gpos is valid and publish separately.

```
rtl_time_estimate_s rtl_time_estimate{};

if (_navigator->home_global_position_valid() && global_position_recently_updated) {
   calc_time_estimate(RTLState::RTL_STATE_NONE, rtl_time_estimate); // pass a ref to rtl_time_estimate to fill the struct
}

// Publish message
rtl_time_estimate.timestamp = hrt_absolute_time();
_rtl_time_estimate_pub.publish(rtl_time_estimate);

Reply:
fair point, see 2bff49475b504b29eba8fb5cf1cb65400ea80482!